lib/fluent/event_router.rb

```
@@ -132,11 +148,35 @@ def match?(tag)
 
     def match(tag)
       collector = @match_cache.get(tag) {
-        find(tag) || @default_collector
+        @default_collector
       }
```

RuboCop (1.75.5)

[convention] Redundant assignment before returning detected.

(Style/RedundantAssignment)